runtime/wasm/src/host_exports.rs

```diff
@@ -812,6 +813,27 @@ pub(crate) fn bytes_to_string(logger: &Logger, bytes: Vec<u8>) -> String {
     s.trim_end_matches('\u{0000}').to_string()
 }
 
+pub(crate) fn abi_encode(params: Vec<Token>) -> Result<Vec<u8>, anyhow::Error> {
+    Ok(encode(&params))
+}
+
+pub(crate) fn abi_decode(mut types: String, data: Vec<u8>) -> Result<Vec<Token>, anyhow::Error> {
+    if !(types.starts_with('(') && types.ends_with(')')) {
+        return Err(anyhow::anyhow!("'ethabi' requires types to start and finish with parentheses, eg: '(uint256,address,(bytes32,int8,address))'"));
+    }
+
+    types.pop();
+    types.remove(0);
```

Comment:
`trim_matches` might be more readable here https://doc.rust-lang.org/std/primitive.str.html#method.trim_matches.

Reply:
Wow, I didn't know about this.
Looks much better, thanks 😊

Reply:
Anyway, this part has been removed, I'm no longer dealing with parenthesis.